calculator/src/main/java/edu/nextstep/camp/calculator/CalculatorViewModel.kt

```diff
@@ -40,8 +49,36 @@ class CalculatorViewModel(
         if (result == null) {
             _errorEvent.emit(Throwable(IncompleteExpressionException()))
         } else {
+            insertExpressionHistory(
+                ExpressionHistory(
+                    expression = expression.value.toString(),
+                    result = result
+                )
+            )
             _expression.value = Expression(listOf(result))
         }
     }
 
+    private fun insertExpressionHistory(history: ExpressionHistory) = viewModelScope.launch {
+        expressionHistoryRepository.insert(history)
+            .onSuccess {}
+            .onFailure { throwable ->
+                _errorEvent.emit(throwable)
+            }
+    }
+
+    fun getAllExpressionHistoryAndShowHistory() = viewModelScope.launch {
+        if (isHistoryVisible.value) {
+            _isHistoryVisible.value = false
+        } else {
```

Comment:
```suggestion
            _isHistoryVisible.value = false
            return@launch
        }
```
return 문을 활용하면, 코드를 좀 더 간결하게 작성할 수 있어요.

Reply:
[92f99a5](https://github.com/next-step/android-calculator-mvvm/pull/113/commits/92f99a50c9ac39d86f84d58540a931a4ba3fea06)